packages/shadydom/src/patch-events.js

```diff
@@ -359,7 +359,38 @@ function fireHandlers(event, node, phase) {
   }
 }
 
-function shadyDispatchEvent(e) {
+// In Chrome 41, `currentTarget` is an own property of every `Event` and its
+// descriptor is a _data_ descriptor and not an accessor descriptor (so there is
+// no getter that could be saved and wrapped) but the browser effectively treats
+// this property as if it had a getter and returns different values on getting
+// as necessary. Further, setting a new descriptor for `currentTarget`, as
+// `shadyDispatchEvent` does below, permanently breaks this special descriptor
+// behavior. In this case, `shadyDispatchEvent` creates a new object with the
+// original event as its prototype (via `Object.create`) and modifies that
+// object's `currentTarget` instead, to avoid breaking the descriptor on the
+// original event.
+//
+// In Chrome 104 and Safari 9, some getters on events do not allow `this` to be
+// anything other than exactly the original event object. This means that the
+// above process of using a proxy-like (not a `Proxy`) object for the event
+// fails because any user that attempts to use properties of the original event
+// will be using the proxy as `this`.
+//
+// Fortunately, these two issues seem to be mutually exclusive amongst the
+// browsers that we test on, so we can specifically check if `currentTarget`
+// exists on a new `Event` and switch based off of that. Safari 9 both doesn't
+// have working descriptors for `Event` properties and doesn't allow getter
+// contexts to be anything other than the original event, so
+// `utils.settings.hasDescriptors` is not a sufficient test for deciding this
```

Comment:
Hmm, so how does this not break Safari 9? Which path does it take?

Reply:
Good catch, my comment is wrong. Safari 9 actually does have working `Event` descriptors, so it's OK to overwrite `currentTarget` on the original event. The real problem is that `utils.settings.hasDescriptors` is false in Safari 9 because it's set like this:

https://github.com/webcomponents/polyfills/blob/08c65f4203b16762b53f220f786fc293828a054a/packages/shadydom/src/utils.js#L31-L33

And Safari 9's `firstChild` descriptor is not configurable.

Reply:
Updated.